modules/ollama/options.go

```diff
@@ -0,0 +1,90 @@
+package ollama
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/docker/docker/api/types/container"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/exec"
+)
+
+var noopCustomizeRequestOption = func(req *testcontainers.GenericContainerRequest) {}
+
+// withGpu requests a GPU for the container, which could improve performance for some models.
+// This option will be automaticall added to the Ollama container to check if the host supports nvidia.
+func withGpu() testcontainers.CustomizeRequestOption {
+	cli, err := testcontainers.NewDockerClientWithOpts(context.Background())
+	if err != nil {
+		return noopCustomizeRequestOption
+	}
+
+	info, err := cli.Info(context.Background())
+	if err != nil {
+		return noopCustomizeRequestOption
+	}
+
+	// if the Runtime does not support nvidia, we don't need to request a GPU
+	if _, ok := info.Runtimes["nvidia"]; !ok {
+		return noopCustomizeRequestOption
+	}
+
+	return testcontainers.WithHostConfigModifier(func(hostConfig *container.HostConfig) {
+		hostConfig.DeviceRequests = []container.DeviceRequest{
+			{
+				Count:        -1,
+				Capabilities: [][]string{{"gpu"}},
+			},
+		}
+	})
+}
+
+// WithModelAndPrompt will run the given model, with a prompt to be passed to the model.
+// If Ollama is not able to run the given model, it will fail to initialise.
+func WithModelAndPrompt(model string, prompt string) testcontainers.CustomizeRequestOption {
+	return withModel(model, prompt)
+}
+
+// WithModel will run the given model, without any prompt.
+// If Ollama is not able to run the given model, it will fail to initialise.
+func WithModel(model string) testcontainers.CustomizeRequestOption {
+	return withModel(model, "")
+}
+
+func withModel(model string, prompt string) testcontainers.CustomizeRequestOption {
+	runCmds := []string{"ollama", "run", model}
```

Comment:
There is a difference between `run` and `pull` command. I think the command should be `pull` here in order to let the container ready with the model in it. WDYT?

Also the `run` command works with models that support chat. For example, it works with `orca-mini` but it doesn't with `all-minilm`

Reply:
@eddumelendez I added the pull option, so that the models are always pull & run.

Besides, I've added the Commit image support (see https://github.com/testcontainers/testcontainers-java/pull/8369), but not forcing users to pass an image but instead generate a random suffix at the end, which is returned in the Commit method

Reply:
Solved in 343194004aee0b24ec331b3ef73d386880755159